File: leetcode75/605. Can Place Flowers/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:

        if n == 0:
            return True

        length = len(flowerbed)

        for i in range(length):

            if flowerbed[i] == 1:
                continue
            
            is_left_empty = (i == 0) or (flowerbed[i - 1] == 0)
            is_right_empty = (i == length - 1) or (flowerbed[i + 1] == 0)

            if is_left_empty and is_right_empty:
                n -= 1
                flowerbed[i] = 1
                if n == 0:
                    return True

        return False

File: leetcode75/605. Can Place Flowers/Solution.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        
        int length = flowerbed.Length;

        for (int i = 0; i < length && n > 0; ++i) {

            if (flowerbed[i] == 1)
                continue;
            
            bool emptyLeftPlot = i-1 < 0 || flowerbed[i-1] == 0;
            bool emptyRightPlot = i+1 >= length || flowerbed[i+1] == 0;

            if (emptyLeftPlot && emptyRightPlot) {
                n = n-1;
                flowerbed[i] = 1;
            }
        }
        return n == 0;
    }
}
